modules/rte_mis_state/rte_mis_state.module

```diff
@@ -209,6 +234,44 @@ function rte_mis_state_form_alter(&$form, FormStateInterface $form_state, $form_
       $form['#validate'][] = 'rte_mis_state_user_location_field_set';
     }
   }
+  // Form alter for the people view page form.
+  elseif ($form_id == 'views_exposed_form' && isset($views) && $views['view']->id() == 'administerusersbyrole_people') {
+    $role_param = \Drupal::request()->query->get('role') ?? NULL;
+    $curr_user_role = \Drupal::currentUser()->getRoles();
+
+    if ($role_param) {
+      if (in_array('state_admin', $curr_user_role)) {
+        if ($role_param == 'district_admin') {
+          $form['role']['#options'] = [
+            'All' => 'Any',
+            'school' => 'School',
+            'school_admin' => 'School Admin',
+            'block_admin' => 'Block Admin',
+            'district_admin' => 'District Admin',
+          ];
+        }
+        elseif ($role_param == 'block_admin') {
+          $form['role']['#options'] = [
+            'All' => 'Any',
+            'school' => 'School',
+            'school_admin' => 'School Admin',
+            'block_admin' => 'Block Admin',
+            'district_admin' => 'District Admin',
+          ];
```

Comment:
@RohanSingh-inno Can you please check if **school**, and **school admin** should be created via `Add user` operation, According to workflow, the school should register the account from the school registration form and their account should not be created by admins.
CC - @kunal-sin

Reply:
This code gives the option to the state admin to list down the users with school and school admin role. While creating a new user we are still not giving the access to state admin to create any **school/school_admin** account.

Reply:
We have modified the permissions. Now State cannot edit **school/ school_admin** accounts.